testing/scenario_app/ios/Scenarios/ScenariosUITests/AccessibilityUITests.m

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/flutter.h>
+#import <XCTest/XCTest.h>
+
+@interface AccessibilityUITests : XCTestCase
+@property(nonatomic, strong) XCUIApplication* application;
+@end
+
+@implementation AccessibilityUITests
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+
+  self.application = [[XCUIApplication alloc] init];
+  self.application.launchArguments = @[ @"--accessibility" ];
+  [self.application launch];
+}
+
+
+- (void)testIfA11yTraversalOrderMatchesAccessibilityScenario{
+
+  NSPredicate *predicate = [NSPredicate predicateWithBlock:^BOOL(id  _Nullable evaluatedObject, NSDictionary<NSString *,id> * _Nullable bindings) {
+    XCUIElement *element = evaluatedObject;
+    return element.label.length > 0;
+  }];
+  XCUIElement *firstElement = [self.application.otherElements elementMatchingPredicate:predicate];
+  if (![firstElement waitForExistenceWithTimeout:30]) {
+    XCTFail(@"Failed due to not able to find any element with 30 seconds");
+  }
+
+  NSLog(@"%@", self.application.debugDescription);
+
+  XCUIElementQuery *query = self.application.otherElements;
+  XCUIElementQuery *allElementsWithA11yLabelQuery = [query matchingPredicate: predicate];
+  NSArray<XCUIElement *> *elements = allElementsWithA11yLabelQuery.allElementsBoundByAccessibilityElement;
```

Comment:
I am not familiar with this method and the documentation on it is spares... Does it actually guarantee that elements are returned in a11y traversal order?

Reply:
As far as I know, the `query` always runs a BFS traversal in the a11y tree. So as long as we only fetch the direct children with the `predicate`, it should be returning the  child elements in the same order as what we set in foo.accessibilityElements.

Reply:
actually it seems like I didn't push the latest commit. Now it should make more sense.